face-detection-explainer.md

```diff
@@ -12,66 +12,129 @@
+partially (or in special cases even fully) outside of the visible image.
+A coordinate is interpreted to represent a location in a normalized square space. The origin of
+coordinates (x,y) = (0.0, 0.0) represents the upper leftmost corner whereas the (x,y) =
+(1.0, 1.0) represents the lower rightmost corner relative to the rendered frame: the x-coordinate (columns) increases rightwards and the y-coordinate (rows) increases downwards.
+
+### Constraints
 
-The constraint `faceDetectionMode` is used by applications to describe the level of facial data that they need. At the lowest enabled level, `presence` will return the sequence of `DetectedFace`, but the `contour` and `landmarks` sequences will be empty. When `faceDetectionMode` is `contour`, arbitrary number of points around the faces will be returned but no landmarks. An user agent might return only four contour points corresponding to face bounding box. If a Web application needs only maximum of four contour points (bounding box), it can set `faceDetectionMode` to `bounding-box` which limits number of contour points to four, residing at the corners of a rectangle around the detected faces.
+New members are added to capabilities, constraints, and settings for Web applications to enable and control face and face landmark detection with `getUserMedia()` and `applyConstraints()` and to query capabilities of face detection with `getCapabilities()` methods. Web applications should not ask more facial metadata than what they need to limit computation. For example, if an applications is content with just a face bounding box, it should set the constraint `humanFaceLandmarkDetectionMode` to `"none"`.
 
-At the highest level, when `faceDetectionMode` is `landmarks`, the full precision contour which is available is returned along with landmark features.
+The enumeration constraints `humanFaceDetectionMode` and `humanFaceLandmarkDetectionMode` set the level of detection needed for human faces and their landmarks, respectively. These settings can be one of the enumeration values in `HumanFaceDetectionMode` and `HumanFaceLandmarkDetectionMode`. When `humanFaceDetectionMode` is `"bounding-box"`, user agent must attempt face detection and set the metadata in video frames correspondingly. When the setting is `"none"`, face description metadata (including landmarks) is not set. Similarly, when `humanFaceLandmarkDetectionMode` is `"none"`, the landmarks (ie. members `leftEye`, `rightEye`, and `mouth` in dictionary `HumanFace`) are not set. When the setting is `"center-point"` and face detection is enabled, the user agent must attempt to detect face landmarks and set the location information in the members of type `HumanFaceLandmark` accordingly. 
 
+If face detection mode is `"none"`, also landmark detection setting must be the same `"none"` because without face detection results the landmark data can not be set either.

```

Comment:
After changing the metadata according to this [issue](https://github.com/w3c/mediacapture-extensions/issues/79) landmark metadata is not anymore syntactically tied to facial metadata and could therefore stored independently. Earlier, facial landmarks were part of the facial metadata and if the latter was not there, the prior couldn't have been either. 

However, even after the metadata syntax change, platform APIs typically don't allow detecting landmarks without also detecting faces. Also I think it would rarely be useful anyway. I think the best is to leave it to the user agent to decide if there is a dependency between the settings `humanFaceDetectionMode` and `humanFaceLandmarkDetectionMode`. I can see here two reasonable behaviors: 1) When `humanFaceLandmarkDetectionMode` is constrained to be `"center-point"`, `humanFaceDetectionMode` is set implicitly to `"bounding-box"` or 2) they can be set independently.

"Landmarks are part of human face and they can be set or not set independently of the whole face bounding box." -- this was inaccurate and misleading. It would have been more correct to say that web application may disable or enable facial landmark detection as long as face metadata is enabled. After the metadata syntax change, the sentence is even less true so changing it into following:

"On many platform APIs, enabling face landmark detection implies also enabling face detection. Therefore the user agent could enable face detection (setting it to `"bounding-box"`) when a web application applies the constraint to enable facial landmark detection. "